library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplayLifecycleCallback.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import com.datadog.android.sessionreplay.processor.SnapshotProcessor
+import com.datadog.android.sessionreplay.recorder.Recorder
+import com.datadog.android.sessionreplay.recorder.ScreenRecorder
+import java.util.WeakHashMap
+
+/**
+ * The SessionReplayLifecycleCallback.
+ * It will be registered as `Application.ActivityLifecycleCallbacks` and will decide when the
+ * activity can be recorded or not based on the `onActivityResume`, `onActivityPause` callbacks.
+ * This is only meant for internal usage and later will change visibility from public to internal.
+ */
+@SuppressWarnings("UndocumentedPublicFunction")
+class SessionReplayLifecycleCallback :
+    Application.ActivityLifecycleCallbacks {
+
+    internal var recorder: Recorder = ScreenRecorder(SnapshotProcessor())
+    internal val resumedActivities: WeakHashMap<Activity, Any?> = WeakHashMap()
```

Comment:
Why we use `WeakHashMap` if we can properly control the reference to the activity by listening to the lifecycle?

Reply:
Same as above